src/components/Pin.vue

```diff
@@ -0,0 +1,38 @@
+<template>
+  <div class="pin-container"
+       @mouseover="isHovered = true"
+       @mouseout="isHovered = false"
+       @click="changeImage">
+    <img :src="pinUrl" alt="Pin" :style="{ transform: isHovered ? 'scale(0.9)' : 'scale(1)', transition: 'transform 0.3s ease', width: '75px', height: '100px' }">
```

Comment:
```suggestion
    <img :src="pinImgs[pinImgsIndex]" alt="Pin" :style="{ transform: isHovered ? 'scale(0.9)' : 'scale(1)', transition: 'transform 0.3s ease', width: '75px', height: '100px' }">
```

(he cambiado pinUrlsIndex por pinImgsIndex en este contexto tmb)